pkg/cmd/cli/cmd/debug.go

```diff
@@ -395,17 +395,36 @@ func (o *DebugOptions) Debug() error {
 }
 
 func (o *DebugOptions) getContainerImageCommand(container *kapi.Container) ([]string, error) {
-	image := container.Image[strings.LastIndex(container.Image, "/")+1:]
-	name, id, ok := imageapi.SplitImageStreamImage(image)
-	if !ok {
-		return nil, errors.New("container image did not contain an id")
+	isi := &imageapi.ImageStreamImport{
+		ObjectMeta: kapi.ObjectMeta{
+			Name: "oc-debug",
+		},
+		Spec: imageapi.ImageStreamImportSpec{
+			Images: []imageapi.ImageImportSpec{
+				{
+					From: kapi.ObjectReference{
+						Kind: "DockerImage",
+						Name: container.Image,
+					},
+					ImportPolicy: imageapi.TagImportPolicy{
+						Insecure: true,
+					},
+				},
+			},
+		},
 	}
-	isimage, err := o.Client.ImageStreamImages(o.Attach.Pod.Namespace).Get(name, id)
+
+	isi, err := o.Client.ImageStreams(o.Attach.Pod.Namespace).Import(isi)
```

Comment:
any dockerCfg secret that is available in the namespace and matches the registry will be used to pull - this is the behaviour of the ImageStreamImages backend.

(by including a build in the extended test and thereby using the authenticated internal registry, this is proven to work).